raxx_elli/lib/raxx/elli/handler.ex

```diff
@@ -30,10 +30,19 @@ defmodule Raxx.Elli.Handler do
     headers = :elli_request.headers(elli_request) |> Enum.map(fn ({k, v}) ->
       {String.downcase(k), String.downcase(v)}
     end)
-    {"Host", authority} = :elli_request.headers(elli_request) |> List.keyfind("Host", 0)
-    [host, port] = String.split(authority, ":")
+
+	[host, port] = 
+	case :elli_request.headers(elli_request) |> List.keyfind("Host", 0) do
+	  {"Host", authority} ->
+		case String.split(authority, ":") do
+			[host, port] -> [host, port]
+			[host] -> [host, "80"] # we need to set the correct port some how...
```

Comment:
Perhaps instead. of `String.split` we should use `URI.parse` on the Host, Gives correct default ports, as well as scheme.

```elixir
URI.parse("http:bob.com").port
80
URI.parse("https:bob.com").port
443
```